## 교통사고 발생량으로부터 인사이트 발견하기

### 문제

다음의 데이터는 2024년도 월별 교통사고 건수이다.
이 데이터에 대해 기초 통계량을 구하고 이를 통해 알 수 있는 사실을 하나 이상
서술하시오


발생월 1 2 3 4 5 6 7 8 9 10 11 12
사고
건수 15129 13836 15738 16300 17605 16690 16252 17000 16956 17720 17455 15668

In [1]:
import numpy as np

In [2]:
data = np.array([15129, 13836, 15738, 16300, 17605, 16690, 16252, 17000, 16956, 17720, 17455, 15668])

In [3]:
#평균,중앙값,범위,표쥰편차
print("평균",np.average(data))
print("중앙값",np.median(data))
print("범위",np.max(data) - np.min(data))
print("표준편차",np.std(data))

평균 16362.416666666666
중앙값 16495.0
범위 3884
표준편차 1089.3134273732037


In [4]:
#절대도수는 월별로 나눠져있음
#상대도수는 각월의 절대 도수 / 전체 도수의 합
data_sum = np.sum(data)
sang = data / data_sum
sang

array([0.07705158, 0.07046636, 0.0801532 , 0.08301545, 0.08966178,
       0.08500171, 0.08277098, 0.08658053, 0.08635644, 0.09024747,
       0.08889783, 0.07979669])

In [7]:
#각 월별 상대 도수
for i in range(len(sang)):
    print("%d월: %.1f%%" % (i+1, sang[i]*100))

1월: 7.7%
2월: 7.0%
3월: 8.0%
4월: 8.3%
5월: 9.0%
6월: 8.5%
7월: 8.3%
8월: 8.7%
9월: 8.6%
10월: 9.0%
11월: 8.9%
12월: 8.0%


In [6]:
print("상대도수 평균:",np.average(sang))

상대도수 평균: 0.08333333333333333


### 결론

상대도수 분석 결과를 보면 10월(9.0%)과 5월(9.0%)의 사고 비율이 상대 도수 평균(8.3%)보다 높았고 2월(7.0%)이 가장 낮게 나타났습니다. 이를 통해 봄이나 가을처럼 야외 활동이 많은 시기에 사고가 자주 일어날 수 있다는 결론을 도출할 수 있었습니다. 야외 활동이 활발해지는 시기에는 정부에서 운전자들에게 경각심을 나타내는 문구(현수막,전광판)를 띄워주는 것이 좋을 것 같습니다.